### PR TITLE
ci(Gha): Add cra to npm smoke test

### LINKED ISSUE #1212: Add create react app to NPM smoke test
There is a new package to add to the publish packages list in NPM smoke test workflow, the CRA template package:

https://www.npmjs.com/package/cra-template-royalnavy

Need to add an additional job that does this to test the consumption of it:

npx create-react-app my-app --template royalnavy

### PR DESCRIPTION
**Pull request by elglup**
## Related issue

Closes #1212 

## Overview

Add create-react-app to NPM smoke test GHA

## Work carried out

Edited npmsmoketest.yml

### CODE REVIEW

File: .github/workflows/npmsmoketest.yml [hunk 1]
```diff
@@ -58,22 +62,16 @@ jobs:
...
-            path: '**/node_modules'
-            key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
-
-
         - name: Publish packages
-          run  : |
+          run: |
             cd main
             yarn install
             npm publish --registry ${{env.LOCAL_REGISTRY}} packages/css-framework
```
Comment (m7kvqbe1) [inline .github/workflows/npmsmoketest.yml:69]: Is it worth setting registry globally so you don't have to explicitly provide the `--registry` argument every time?

```
npm config set registry ${{env.LOCAL_REGISTRY}}
```

**Review by m7kvqbe1 — CHANGES_REQUESTED** [review]
LGTM - I've just left one comment - maybe there is a reason to not do what i've suggested?

File: .github/workflows/npmsmoketest.yml (hunk 1 above)
Comment (elglup) [reply .github/workflows/npmsmoketest.yml:69]: We could although in the next job we need to run yarn install again (not using local registry), so would need to set & unset

Comment (m7kvqbe1) [reply .github/workflows/npmsmoketest.yml:69]: Ok fair enough - feel free to just leave it as is then

**Review by m7kvqbe1 — APPROVED** [review]

**Pull request merged**